lua/autorun/server/sv_restrict_propcore.lua

```diff
@@ -22,24 +22,46 @@ function restrictPropCoreFunctions()
         --"propBreak(e:)"
     }
 
-    for _, signature in pairs( restrictedFunctions ) do
-        if wire_expression2_funcs then
-            local oldFunc = wire_expression2_funcs[signature][3]
-
-            wire_expression2_funcs[signature][3] = function( self, ... )
-                if disallowedRanks[self.player:GetUserGroup()] == nil then
-                    local isInBuildMode = self.player:GetNWBool("CFC_PvP_Mode", false) == false
-
-                    if isInBuildMode or self.player:IsAdmin() then
-                        return oldFunc( self, ... )
-                    else
-                        self.player:ChatPrint( "You can't use PropCore in PvP mode" )
-                    end
-                else
-                    self.player:ChatPrint( "You don't have access to " .. signature )
-                end
+
+    local adminOnlyFunctions = {
+        "use(e:)"
+    }
+
+    restrict( restrictedFunctions, function( self,  ... )
+        local isInBuildMode = self.player:GetNWBool( "CFC_Pvp_Mode" ) == false
+
+        if disallowedRanks[self.player:GetUserGroup()] then
+            return false, "You don't have access to this function"
+        elseif isInBuildMode and not self.player:IsAdmin() then
+             return false, "You can't use propcore in PvP"
+        else
+            return true
+        end
```

Comment:
```suggestion
        if disallowedRanks[self.player:GetUserGroup()] then
            return false, "You don't have access to this function"
        end
        
        isInBuildMode and not self.player:IsAdmin() then
            return false, "You can't use propcore in PvP"
        end
```

This would simplify the code, make it easier to read, and allow for more checks to be implemented.